deepchem/feat/smiles_tokenizer.py

```diff
@@ -0,0 +1,183 @@
+# The vocab may be expanded in the near future
+
+import collections
+import logging
+import os
+import re
+import numpy as np
+import pkg_resources
+from typing import List
+from transformers import BertTokenizer
+
+# export
+SMI_REGEX_PATTERN = r"(\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|\(|\)|\.|=|#|-|\+|\\|\/|:|~|@|\?|>>?|\*|\$|\%[0-9]{2}|[0-9])"
+
+def get_default_tokenizer():
+    default_vocab_path = (
+        pkg_resources.resource_filename(
+            "chemberta",
+            "tokenizers/vocab.txt"
+        )
+    )
+    return SmilesTokenizer(default_vocab_path)
+
+class SmilesTokenizer(BertTokenizer):
+    r"""
+    Constructs a SmilesTokenizer.
+    Bulk of code is from https://github.com/huggingface/transformers and https://github.com/rxn4chemistry/rxnfp
+    Args:
+        vocab_file: Path to a SMILES character per line vocabulary file
+    """
+
+    def __init__(
+            self,
+            vocab_file='',
+            # unk_token="[UNK]",
+            # sep_token="[SEP]",
+            # pad_token="[PAD]",
+            # cls_token="[CLS]",
+            # mask_token="[MASK]",
+            **kwargs
+    ):
+        """Constructs a BertTokenizer.
+        Args:
+            **vocab_file**: Path to a SMILES character per line vocabulary file
+        """
+        super().__init__(vocab_file, **kwargs)
+        # take into account special tokens in max length
+        self.max_len_single_sentence = self.max_len - 2
+        self.max_len_sentences_pair = self.max_len - 3
+
+        if not os.path.isfile(vocab_file):
+            raise ValueError(
+                "Can't find a vocab file at path '{}'.".format(vocab_file)
+            )
+        self.vocab = load_vocab(vocab_file)
+        self.highest_unused_index = max(
+            [
+                i for i, v in enumerate(self.vocab.keys())
+                if v.startswith("[unused")
+            ]
+        )
+        self.ids_to_tokens = collections.OrderedDict(
+            [(ids, tok) for tok, ids in self.vocab.items()]
+        )
+        self.basic_tokenizer = BasicSmilesTokenizer()
+        self.init_kwargs["max_len"] = self.max_len
+
+    @property
+    def vocab_size(self):
+        return len(self.vocab)
+
+    @property
+    def vocab_list(self):
+        return list(self.vocab.keys())
+
+    def _tokenize(self, text):
+        split_tokens = [token for token in self.basic_tokenizer.tokenize(text)]
+        return split_tokens
+
+    def _convert_token_to_id(self, token):
+        """ Converts a token (str/unicode) in an id using the vocab. """
+        return self.vocab.get(token, self.vocab.get(self.unk_token))
+
+    def _convert_id_to_token(self, index):
+        """Converts an index (integer) in a token (string/unicode) using the vocab."""
+        return self.ids_to_tokens.get(index, self.unk_token)
+
+    def convert_tokens_to_string(self, tokens):
```

Comment:
Could you switch to Numpy doc style and add type annotations for these other methods as well?

Reply:
Type annotation still missing here and on some of the other methods